File: app/src/main/java/com/drivingschoolrwandaapp/models/entities/TestResult.java
package com.drivingschoolrwandaapp.models.entities;

public class TestResult {

    private final int score;
    private final int totalMarks;
    private final boolean passed;

    public TestResult(int score, int totalMarks, boolean passed) {
        this.score = score;
        this.totalMarks = totalMarks;
        this.passed = passed;
    }

    public int getScore() {
        return score;
    }

    public int getTotalMarks() {
        return totalMarks;
    }

    public boolean isPassed() {
        return passed;
    }
}

File: app/src/main/java/com/drivingschoolrwandaapp/models/entities/TestResult.kt
package com.drivingschoolrwandaapp.models.entities

import com.google.gson.annotations.SerializedName

data class TestResult(
    val score: Int,
    val totalMarks: Int,
    @get:JvmName("isPassed") val passed: Boolean,
    val testNumber: Int = 0,
    val testName: String = "",
    val testId: Int = 0,
    val date: Long = 0,
    val duration: Int = 0,
    val elapsedSeconds: Int = 0,
    val correctCount: Int = 0,
    val wrongCount: Int = 0,
    val skippedCount: Int = 0
)
